internal/component/otelcol/extension/jaeger_remote_sampling/internal/jaegerremotesampling/config_test.go

```diff
@@ -27,9 +27,9 @@ func TestLoadConfig(t *testing.T) {
 		{
 			id: component.NewID(typeStr),
 			expected: &Config{
-				HTTPServerConfig: &confighttp.ServerConfig{Endpoint: ":5778"},
+				HTTPServerConfig: &confighttp.ServerConfig{Endpoint: "localhost:5778"},
```

Comment:
I think this might be surprising to users, as suddenly telemetry might be not be able to get pushed anymore.

Looking at the [upstream docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/jaegerremotesampling/README.md) I think the default is 0.0.0.0 for this reason. Can we align it with that value?

Once we start supporting OTel Collector feature gates (eg. such as `component.UseLocalHostAsDefaultHost`) then it might make sense to also support this.

Reply:
@tpaschalis If you look at [this discussion](https://github.com/open-telemetry/opentelemetry-collector/issues/8510) upstream, they're working towards changing the default to 127.0.0.1 in a future release. Currently if you enable a specific feature gate (`UseLocalHostAsDefaultHost`) then all the components default to 127.0.0.1 instead.

I think we need to figure out if we're going to do this before 1.0, since it would be a breaking change when it does get implememnted.

Reply:
Since this is pretty breaking, maybe we can park this one and accept the deviance as part of `grafana/agent` features. Not sure how we can reconcile with upstream for this one.